src/newcharacter.cpp

```diff
@@ -1049,14 +1049,23 @@ tab_direction set_stats( avatar &u, points_left &points )
     } while( true );
 }
 
+static struct {
+    /** @related player */
+    bool operator()( const trait_id *a, const trait_id *b ) {
+        return std::abs( a->obj().points ) > std::abs( b->obj().points );
+    }
+} traits_sorter;
+
```

Comment:
I feel this can be a lambda function object inside `set_traits` where it is used.

Reply:
I wrote it like that so it could match `scenario_sorter` and `profession_sorter`. I think it might be easier for someone in the future to understand the code if they are similar.